index.js

```diff
@@ -53,11 +56,59 @@ var InvalidJoinSourceError = TypedError({
     actual: null
 });
 
+var AppRequiredError = TypedError({
+    type: 'ringpop.options-app.required',
+    message: 'Expected `options.app` to be a non-empty string.\n' +
+        'Must specify an app for ringpop to work.\n'
+});
+
+var OptionsRequiredError = TypedError({
+    type: 'ringpop.options.required',
+    message: 'Expected `options` argument to be passed.\n' +
+        'Must specify options for `RingPop({ ... })`.\n'
+});
+
+var HostPortRequiredError = TypedError({
+    type: 'ringpop.options-host-port.required',
+    message: 'Expected `options.hostPort` to be valid.\n' +
+        'Got {hostPort} which is not {reason}.\n' +
+        'Must specify a HOST:PORT string.\n',
+    hostPort: null,
+    reason: null
+});
+
 function RingPop(options) {
     if (!(this instanceof RingPop)) {
         return new RingPop(options);
     }
 
+    if (!options) {
```

Comment:
SO MUCH CODEZ.

_required_ options always rubs me the wrong way. It's a contradiction. Rename to `params` now that we're strictly enforcing. Thoughts?